source/code/plugins/VMInsightsDataCollector.rb

```diff
@@ -99,17 +99,12 @@ def get_number_of_cpus
         def get_filesystems
             result = []
             df = File.join(@root, "bin", "df")
-            if @root != '/'
-                cmd = "#{df} --block-size=1 -T"
-            else
-                cmd = "#{df} --block-size=1 -T | awk '{print $2, $1, $7, $3, $5}'"
-            end
-            IO.popen(cmd, { :in => :close }) { |io|
+            IO.popen([df, "--block-size=1", "-T"], { :in => :close, :err => File::NULL }) { |io|
                 while (line = io.gets)
-                    if (line =~ /ext[234]/)
+                    a = line.split(" ")
+                    if (a[1] =~ /ext[234]/)
```

Comment:
This is definitely an improvement over what I had originally.

Looking at it with the benefit of time away from it, please add to the mocked test case file system types "text2" and "ext3a" to make sure the product doesn't include them.

Reply:
Will the df command ever return filesystem type that is malformed like text2 or ext3a?